docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -1,466 +1,1741 @@
+   [...]
+
+..
+
+.. note::
+   The above command continues to run until terminated. You can
+   terminate it with ``Ctrl-C`` at any time.
+
+Auth
+~~~~
+
+If you are running PQS against a Participant that has auth, you must
+provide some credentials. For example:
+
+.. code:: bash
+
+   $ ./scribe.jar pipeline ledger postgres-document \
+       --source-ledger-auth OAuth \
+       --pipeline-oauth-clientid my_client_id \
+       --pipeline-oauth-clientsecret deadbeef \
+       --pipeline-oauth-cafile ca.crt \
+       --pipeline-oauth-endpoint https://my-auth-server/token
+
+For more information see the `Authentication <#auth>`__ section.
+
+Database
+========
 
-Relational databases excel at storing structured data for which the schema is
-known in advance. However, they have traditionally lacked mechanisms for data
-that is more dynamic or evolves. For example, you may want to store arbitrary
-Daml contracts and might prefer not to update the database schema every time
-the underlying template changes.
+PQS supports a single datastore format known as ``postgres-document``.
+This uses PostgreSQL to store the data in a document-oriented
+(schemaless) way, making extensive use of JSONB.
 
-PostgreSQL helps manage unstructured data through native support for JSON data
-and allows queries to process this data. For best performance, the PQS stores
-data as JSONB only.
+Ledger Structure
+----------------
 
-An example query might look like this:
+To understand how PQS stores data, you need to understand the ledger
+data model. In simple terms, the Daml ledger is composed of a sequence
+of transactions, which contain events. Events can be:
 
-.. code-block:: sql
+-  Creates: creation of contracts / interface views.
+-  Exercise: of a choice of contracts / interface views.
+-  Archive: end of the lifetime of contracts / interface views.
 
-    SELECT *
-    FROM contract
-    WHERE payload->>'isin' = 'abc123'
-    ORDER BY payload->'issuanceData'->'issueDate'->>'Some';
+Ledger Data Scope
+-----------------
 
-For more information on querying JSON data, see the section `JSON Functions
-and Operators <https://www.postgresql.org/docs/12/functions-json.html>`__ in
-the PostgreSQL manual. The operators ``->``, ``->>``, ``#>``, ``#>>``, and
-``@>`` may be of particular interest.
+PQS can run in two modes as specified by the ``--pipeline-datasource``
+configuration. The following table shows the differences between the two
+modes, in terms of data availability via the respective `Read
+API <#querying>`__ functions :
+
++----------------------+----------------------+----------------------+
+| Mode                 | TransactionStream(1) | T                    |
+|                      |                      | ransactionTreeStream |
++======================+======================+======================+
+| Transient events     | ✗                    | ✓                    |
```

Comment:
I really like the table. A very nice breakdown.  I do have a couple of comments:
- I didn't see a definition of a "Transient event" in this user guide or in docs.daml.com.  Could we please add one?  I assume this is more about ephemeral contracts within a transaction.
- Can we indicate that one of these needs to be selected and can't be changed without dropping the PQS DB?  Using the phrase "Opt-in" suggests that this is an add-on and not mutually exclusive.
- The subsript for TransactionStream(1) (i.e., the '(1)') didn't make it through.  Not sure what it refers to.

Reply:
Fixed. We can't indicate what can't be changed - because you really can't change anything (template filter, party filter, Transaction Stream Type). Any change, based on the Ledger API requires a re-seed.